portfolio/src/test/java/com/google/sps/DataServletTest.java

```diff
@@ -0,0 +1,67 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps;
+
+import static org.mockito.Mockito.*;
+
+import com.google.appengine.tools.development.testing.LocalDatastoreServiceTestConfig;
+import com.google.appengine.tools.development.testing.LocalServiceTestHelper;
+import com.google.sps.servlets.*;
+import java.io.PrintWriter;
+import java.io.StringWriter;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public final class DataServletTest {
+  private final LocalServiceTestHelper helper =
+      new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+  @Before
+  public void setUp() {
+    helper.setUp();
+  }
+
+  @After
+  public void tearDown() {
+    helper.tearDown();
+  }
+
+  @Test
+  public void testJSONConverter() throws Exception {
```

Comment:
Here's a giant comment for you (feel free to follow up with me offline as necessary):

- I think this is outside the scope of the initial exercises - but since we're going into uncharted territory - can you switch your assertions to use [Google's truth library](https://truth.dev/)? This is what we'll be using for the final project.
  - You'll need to add a POM dependency to grab this library.
- This test is testing two behaviors - would be better if we could isolate and augment:
  - On the POST side (let's use the DatastoreService to verify the results instead of our API):
    - A test that verifies that a single comment is persisted
    - A test that verifies that multiple comments are persisted
    - A test that verifies behavior when `text-input` is null
    - Verification that `sendRedirect` is invoked as expected
  - On the GET side (let's use the DatastoreService to seed data for test):
    - Verify that a single comment is returned
    - Verify that multiple comments are returned
    - Verify that an empty datastore returns an empty response
    - Verify that the content type is correctly set.

Feel free to stack chain PRs here or just keep things as a single CL. I'm excited about these tests!

Reply:
this is a great comment thanks a bunch! I'll start working on creating more tests/incorporating Google's truth library! How much should I change of the current test for this CL (I was thinking of chaining PRs)?

Reply:
I'm fine with all the tests in a single CL. I think it makes sense as a logical unit.

Reply:
ok awesome! thanks!

Reply:
how do you verify that sendRedirect is invoked properly? is it just checking the response content type?

Reply:
Have a look at https://site.mockito.org/ - you can use mockito to verify that a method was invoked with specific arguments on your mock object.